pkg/controller/metrics/metrics.go

```diff
@@ -398,44 +414,57 @@ func (mc *Calculator) Start(ctx context.Context) error {
+		// Histograms
 		case metricsconfig.CurrentStopping:
 			metrics.Registry.MustRegister(MetricStoppingClustersSeconds)
-			mapMetricToDuration[MetricStoppingClustersSeconds] = metric.Duration.Duration
+			mapMetricToDurationHistograms[MetricStoppingClustersSeconds] = metric.Duration.Duration
 		case metricsconfig.CurrentResuming:
 			metrics.Registry.MustRegister(MetricResumingClustersSeconds)
-			mapMetricToDuration[MetricResumingClustersSeconds] = metric.Duration.Duration
+			mapMetricToDurationHistograms[MetricResumingClustersSeconds] = metric.Duration.Duration
 		case metricsconfig.CurrentWaitingForCO:
 			metrics.Registry.MustRegister(MetricWaitingForCOClustersSeconds)
-			mapMetricToDuration[MetricWaitingForCOClustersSeconds] = metric.Duration.Duration
+			mapMetricToDurationHistograms[MetricWaitingForCOClustersSeconds] = metric.Duration.Duration
 		case metricsconfig.CumulativeHibernated:
 			metrics.Registry.MustRegister(MetricClusterHibernationTransitionSeconds)
-			mapMetricToDuration[MetricClusterHibernationTransitionSeconds] = metric.Duration.Duration
+			mapMetricToDurationHistograms[MetricClusterHibernationTransitionSeconds] = metric.Duration.Duration
 		case metricsconfig.CumulativeResumed:
 			metrics.Registry.MustRegister(MetricClusterReadyTransitionSeconds)
-			mapMetricToDuration[MetricClusterReadyTransitionSeconds] = metric.Duration.Duration
+			mapMetricToDurationHistograms[MetricClusterReadyTransitionSeconds] = metric.Duration.Duration
+		// Gauges
+		case metricsconfig.CurrentClusterSyncFailing:
+			metrics.Registry.MustRegister(MetricClusterSyncFailingSeconds)
+			mapMetricToDurationGauges[MetricClusterSyncFailingSeconds] = metric.Duration.Duration
 		}
 	}
 }
 
-// ShouldLogDurationMetric decides whether the corresponding duration metric should be logged. It first checks if that
-// optional metric has been opted into for logging, and the duration has crossed the threshold set
-func ShouldLogDurationMetric(metricToLog *prometheus.HistogramVec, timeToLog float64) bool {
-	duration, ok := mapMetricToDuration[metricToLog]
+// ShouldLogHistogramDurationMetric decides whether the corresponding duration metric of type histogram should be logged.
+// It first checks if that optional metric has been opted into for logging, and the duration has crossed the threshold set.
+func ShouldLogHistogramDurationMetric(metricToLog *prometheus.HistogramVec, timeToLog float64) bool {
+	duration, ok := mapMetricToDurationHistograms[metricToLog]
+	return ok && duration.Seconds() <= timeToLog
+}
+
+// ShouldLogGaugeDurationMetric decides whether the corresponding duration metric of type gauge should be logged. It
+// first checks if that optional metric has been opted into for logging, and the duration has crossed the threshold set.
+func ShouldLogGaugeDurationMetric(metricToLog *prometheus.GaugeVec, timeToLog float64) bool {
+	duration, ok := mapMetricToDurationGauges[metricToLog]
 	return ok && duration.Seconds() <= timeToLog
 }
```

Comment:
NB: I think we'll be able to collapse the maps and these funcs using generics once https://github.com/openshift/hive/pull/1803 lands :)